### PR TITLE
Features/importable

### PR DESCRIPTION
**Pull request by brettviren**
This lets spack.* modules be importable into a bare Python.  The `lib/spack/spack/tests/importing.sh` will run through all modules to test they can be imported without error.  For in-source testing, `PYTHONPATH` must add `lib/spack`.

This also moves the backport and override modules to be under `spack.util`.  

Note, this does not address the location variables in `spack/__init__.py`.

### CODE REVIEW

File: lib/spack/spack/test/importing.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+export PYTHONPATH=$libdir
+
+set -e
+set -x
+python -c 'import spack'
+
+
+cd $topdir
+
+for mod in $(ls lib/spack/spack/*/__init__.py|awk -F/ '{print $4}')
```
Comment (mathstuf) [inline lib/spack/spack/test/importing.sh]: Ick. Parsing `ls`. How about:

```sh
for mod in lib/spack/spack/*/; do
    [ -r "$mod/__init__.py" ] && \
        python -c "import spack.$mod"
done
```

Comment (mathstuf) [inline lib/spack/spack/test/importing.sh]: Oops. Should use `basename`.

**Comment by adamjstewart** [comment]
Is this PR still active?

**Comment by brettviren** [comment]
Not on my end.

**Comment by adamjstewart** [comment]
Like do you still want this to be merged? If so it needs to be rebased. What is the use case for being able to import each of these files?

**Comment by brettviren** [comment]
> Like do you still want this to be merged? 

4 months ago I did.  Right now, I don't intend to pursue it.

> What is the use case for being able to import each of these files?

Maybe you want to subscribe to the mailing list.  This topic has come up several times and most recently about a week ago.

**Comment by adamjstewart** [comment]
The Google Group has a lot of topics. If there are some in particular that this issue addresses, you may want to link to them in your PR.

If you want this PR to be merged, it needs to be rebased. If you don't want it to be merged anymore, it should probably be closed. Just trying to clean up and merge old PRs that have been left behind :smiley:

**Pull request closed without merging**